README.md

```diff
@@ -25,9 +25,41 @@ To run the program, you must enter the command `cargo run --bin <binary_name>` a
 
 ## Testing
 
-There are two different sets of tests: unit tests and end-to-end (e2e tests). To run unit tests, use `cargo test`. 
-To run end-to-end testing, use `make e2e-test`, which will create a test user, ssh directory, place mock ssh keys, and 
-then clean up the test artifacts afterwards.
+Azure-init includes two types of tests: unit tests and end-to-end (e2e) tests.
+
+### Running Unit Tests
+
+To run unit tests, use the following commands based on the scope of testing:
+
+1. **Repository-Wide Unit Tests:**
+From the root directory of the repository, run:
+
+```
+cargo test
+```
+
+This will execute the unit tests defined in the top-level binaries and modules but will **not** include tests from submodules like libazureinit.
+
+2. **Unit Tests in** `libazureinit`:
+
+To run the full set of unit tests, including those within the `libazureinit` library, navigate to the `libazureinit` directory and run:
+```
+cd libazureinit
+cargo test
+```
```

Comment:
Shame on me for not updating this in #96. We could document here that users should run `cargo test --workspace` to pick up all the tests.

Reply:
Ah, perfect! That's an easy fix.